sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -3270,13 +3269,14 @@ def _UpdateSessionIfRequired(
             self.session.update_session(response_result, response_headers)
 
     def _get_partition_key_definition(self, collection_link: str) -> Optional[Dict[str, Any]]:
-        partition_key_definition = None
+        partition_key_definition: Optional[Dict[str, Any]]
         # If the document collection link is present in the cache, then use the cached partitionkey definition
-        if collection_link in self.partition_key_definition_cache:
-            partition_key_definition = self.partition_key_definition_cache.get(collection_link)
+        if collection_link in self.container_properties_cache:
+            cached_collection: Dict[str, Any] = self.container_properties_cache.get(collection_link, {})
+            partition_key_definition = cached_collection.get("partitionKey")
         # Else read the collection from backend and add it to the cache
         else:
             collection = self.ReadContainer(collection_link)
             partition_key_definition = collection.get("partitionKey")
-            self.partition_key_definition_cache[collection_link] = partition_key_definition
+            self.container_properties_cache[collection_link] = collection
```

Comment:
should this be _set_properties_cache(collection)?

Reply:
yes, thank you for catching that.